models/userDao.js

```diff
@@ -0,0 +1,57 @@
+const appDataSource = require("./appDataSource");
+
+const createUser = async (email, name, password, address, phone, birthdate) => {
+  try {
+    return appDataSource.query(
+      `INSERT INTO users(
+        name,
+        password,
+        address,
+        phone,
+        email,
+        birthdate
+        ) VALUES (?, ?, ?, ?, ?, ?);
+  `,
+      [name, password, address, phone, email, birthdate]
+    );
+  } catch {
+    const error = new Error("INVALID_DATA_INPUT");
+    error.statusCode = 500;
+    throw error;
+  }
+};
+
+const getUserByEmail = async (email) => {
+  try {
+    const user = appDataSource.query(
+      `SELECT 
+      email,
+      password
+      FROM 
+      users 
+      WHERE email = ?;`,
+
+      [email]
+    );
+    return user;
+  } catch {
+    const error = new Error("INVALID_DATA_INPUT");
+    error.statusCode = 500;
+    throw error;
+  }
+};
+
+const checkMail = async (email) => {
+  const [emailC] = appDataSource.query(
+    `SELECT * FROM users WHERE email = ?;`,
+
+    [email]
+  );
+  return emailC;
+};
```

Comment:
이메일을 통해 단순 조건에 일치하는 데이터가 있냐 없냐만 따질 때는 모든 컬럼을 불러올 필요가 없습니다. 그리고 API 코드로서 raw query를 쓸 때는 애스터리스크(`*`)를 쓰지 않아야 합니다. id 혹은 어떤 컬럼이라도 단 하나만 불러와도 데이터가 있냐 없냐의 유무 판단에는 충분합니다.

덧붙여 지금같이 단순 존재 여부만을 파악할 때에는 db query를 통해 실제 어떠한 데이터를 받아오는 것 보다 SELECT EXIST 를 활용하여 참, 거짓만 판단하는게 보다 더 효율적입니다. (MySQL에서 참과 거짓은 boolean이 아닌 1, 0으로 리턴됩니다.)

```suggestion
const checkRegisterdEmail = async (email) => {
  const [result] = appDataSource.query(
    `SELECT EXIST (
        SELECT
          id
        FROM users
        WHERE email = ?
      ) as registerd`,
    [email]
  );
  return !!parseInt(result.registerd);        // 0, 1로 나타내어지는 result.registerd를 double bang (!!)을 붙여 booltype으로 변환하여 return합니다. 
};
```

Reply:
넵 수정해서 기능작동상태까지 확인했습니다 !